pkg/generate/app/imagestreamlookup.go

```diff
@@ -56,10 +58,20 @@ func (r ImageStreamSearcher) Search(terms ...string) (ComponentMatches, error) {
 				} else {
 					imageref, _ := imageapi.ParseDockerImageReference(term)
 					imageref.Name = stream.Name
+
+					localSearchTag := searchTag
+					if specTag, ok := stream.Spec.Tags[searchTag]; ok && followTag {
```

Comment:
// "if the user didn't specify a specific tag (so we're using latest), and the "latest" imagestreamtag is actually just 
// following another imagestreamtag, then use the actual imagestreamtag that "latest" is following as the match 
// result instead of the "latest"imagestream tag".  Also only do this if the name IST that "latest" is following does 
// not contain a ":" because....???

at least I think that's what this is doing, i'd like to save the next person who hits this code the trouble of trying to figure it out.